integration/Substra/plot_substrafl_flamby.py

```diff
@@ -0,0 +1,438 @@
+    Baseline as baseline_model,
+    BaselineLoss as baseline_loss,
+    Optimizer as optimizer,
+    BATCH_SIZE as batch_size,
+    LR as ixi_lr,
+)
+
+
+model = baseline_model()
+optimizer = optimizer(model.parameters(), lr=ixi_lr)
+criterion = baseline_loss()
+
+
+# ## Substrafl imports
+#
+
+# In[10]:
+
+
+from typing import Any
+
+from substrafl.algorithms.pytorch import TorchFedAvgAlgo
+from substrafl.dependency import Dependency
+from substrafl.strategies import FedAvg
+from substrafl.nodes import AggregationNode
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.index_generator import NpIndexGenerator
+from substrafl.experiment import execute_experiment
+
+import torch
+
+
+# ## Substrafl algo definition
+#
+# To instantiate a Substrafl [**Torch Algorithm**](https://docs.substra.org/en/latest/substrafl_doc/api/algorithms.html#torch-algorithms), you need to define a torch Dataset with a specific `__init__` signature, that must contain (self, x, y, is_inference). This torch Dataset is useful to preprocess your data on the `__getitem__` function.
+#
+# This dataset is passed **as a class** to the [**Torch Algorithms**](https://docs.substra.org/en/latest/substrafl_doc/api/algorithms.html#torch-algorithms).
+# Indeed, this torch Dataset will be instantiated within the algorithm, using the opener functions as x and y parameters.
+#
+# The index generator will be used a the batch sampler of the dataset, in order to save the state of the seen samples during the training, as Federated Algorithms have a notion of `num_updates`, which forced the batch sampler of the dataset to be stateful.
+#
+# Number of model update between each FL strategy aggregation.
+
+# In[11]:
+
+
+NUM_UPDATES = 16
+
+index_generator = NpIndexGenerator(
+    batch_size=batch_size,
+    num_updates=NUM_UPDATES,
+)
+from flamby.datasets.fed_tcga_brca import FedTcgaBrca
+
+
+class TorchDataset(FedTcgaBrca):
+    def __init__(self, x, y, is_inference):
+        config = x
+        super().__init__(**config)
+
+
+class MyAlgo(TorchFedAvgAlgo):
+    def __init__(self):
+        super().__init__(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            index_generator=index_generator,
+            dataset=TorchDataset,
+            seed=42,
+        )
+
+    def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_path):
+
+        predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._index_generator.batch_size)
+
+        self._model.eval()
+
+        predictions = torch.Tensor([])
+        with torch.inference_mode():
+            for x, _ in predict_loader:
+                x = x.to(self._device)
+                predictions = torch.cat((predictions, self._model(x)), 0)
+
+        predictions = predictions.cpu().detach()
+        self._save_predictions(predictions, predictions_path)
+
+        return predictions
```

Comment:
In the default local predict, we assume that the __getitem__ of the dataset take into account the 'is_inference' argument and only return x. But as the __getitem__ is the one from flamby, the is_inference in used here. So we need to overwrite the _local_predict just for `for x, _ in predict_loader:` instead of `for x in predict_loader:`

Reply:
Thank you for the explanation!
Can you add a small comment to explain this?